### ISSUE TITLE
Tables don't fail over

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4201]
I created a three-node cluster like this:
```
./rethinkdb --bind all -d data_0 -o 0 &
./rethinkdb --bind all -d data_1 -o 1 -j localhost &
./rethinkdb --bind all -d data_2 -o 2 -j localhost &
```

Then created a table and configured it to 3 replicas:
```
r.tableCreate('t1')
r.table('t1').reconfigure({replicas: 3, shards: 1})
```

`r.table('t1').status()` confirmed that all replicas were ready.

Then I shut down the current primary. After 2-3 minutes they remaining two replicas hadn't elected a new primary. `r.table('t1').status()` looked as follows:
```
{
    "db": "test" ,
    "id": "b9821a78-ce8f-4c9c-a5ad-44c9611af287" ,
    "name": "t2" ,
    "shards": [
        {
            "primary_replicas": [ ],
            "replicas": [
                {
                    "server": "sinister_lbf" ,
                    "states": [
                        "waiting_for_primary_replica"
                    ]
                } ,
                {
                    "server": "sinister_zpd" ,
                    "states": [
                        "disconnected"
                    ]
                } ,
                {
                    "server": "sinister_x7p" ,
                    "states": [
                        "waiting_for_primary_replica"
                    ]
                }
            ]
        }
    ] ,
    "status": {
        "all_replicas_ready": false ,
        "ready_for_outdated_reads": true ,
        "ready_for_reads": false ,
        "ready_for_writes": false
    }
}
```

It happened two out of two times I tried.

@timmaxw this should work right? Is this a bug?

**Comment by timmaxw** [comment]
Yes, definitely a bug.

**Comment by timmaxw** [comment]
It looks like the secondaries are not reporting `failover_timeout_elapsed=true` if the contract's `primary` is not set. It's an unintended side effect of 01f7026b28b9f1876fd7b557ba6eeaeb26063171. It should be pretty easy to fix.

**Comment by timmaxw** [comment]
Merged into `raft` as of c05d40aaf69478c29a497d30bc2919054942d016.
